code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1324,3 +1324,232 @@
 	var/mob/living/carbon/xenomorph/X = owner
 	X.hive.purchases.interact(X)
 	return succeed_activate()
+
+// ***************************************
+// *********** Tail Stab
+// ***************************************
+//totally not stolen from punch code
+/datum/action/ability/activable/xeno/tail_stab
+	name = "Tail Stab"
+	action_icon_state = "tail_attack"
+	desc = "Strike a target within two tiles with a sharp tail for armor-piercing damage, stagger and slowdown. Deals double AP, damage, stagger and slowdown to grappled targets, structures and machinery."
+	ability_cost = 30
+	cooldown_duration = 10 SECONDS
+	use_state_flags = ABILITY_USE_AGILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_TAIL_STAB,
+	)
+	target_flags = ABILITY_MOB_TARGET|ABILITY_MOB_TARGET
+	var/range = 2
```

Comment:
autodoc

Reply:
what about autodoc?

Reply:
Auto-documentation.
You use /// on the line before your var/VARNAME
Your autodoc should be informative enough that anyone can read it and understand what the var is for.
You also do the same whenever you create a new proc, so as to describe what your proc does for anyone who reads the code.

Reply:
ah i literally added a damn exception for the autodoc machine thinking you meant itd damage damn autodocs LMAO